### DISCUSSION TITLE
What's the point of the event listeners not being attached to the actual elements

### DISCUSSION THREAD

**Discussion by danieltroger** [solidjs/solid#630] [Q&A] [answered]
And overriding event.currentTarget to fake that they are?

please answer "dumb question" if I should stop asking this. I'm just curious and thought I'd just as well ask

**Comment by atk** [comment, marked as answer]
Short answer:  performance.

Slightly longer answer: running the full DOM event queue with bubbling and capturing is taking some time. If instead you directly subscribe to deferred events, you're basically jumping over those parent nodes that do not subscribe to it.

**Reply by danieltroger** [reply]
Ok, thanks. So it doesn't matter if I put it onto the elements directly sometimes.
